### PR TITLE
jupp 3.1jupp40 (new formula)

### PR DESCRIPTION
**Pull request by johnsonjh**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Add jupp 3.1jupp40

### CODE REVIEW

File: Formula/jupp.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  uses_from_macos "ncurses"
+  on_macos do
+    depends_on "gnu-sed" => :build
+  end
+
+  def install
+    if OS.mac?
+      inreplace "Makefile.am", "sed ", "#{Formula["gnu-sed"].opt_bin}/gsed"
+      inreplace "Makefile.in", "sed ", "#{Formula["gnu-sed"].opt_bin}/gsed"
+    end
```
Comment (SMillerDev) [inline Formula/jupp.rb:23]: Why not just add gnubin to the path?

File: Formula/jupp.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    system "autoreconf", "-vfi"
+    system "sh", "./configure", "--prefix=#{prefix}",
```
Comment (SMillerDev) [inline Formula/jupp.rb:25]:
```suggestion
    system "sh", "./configure", *std_configure_args,
```

File: Formula/jupp.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+                                "--enable-sysconfjoesubdir=/jupp",
+                                "--disable-dependency-tracking"
+    system "make", "install"
+  end
+
+  test do
+    assert_match "File (Unnamed) not changed so no update needed.",
+      shell_output("echo 'spawn #{bin}/jupp;send \"
q\";expect eof'|env TERM=tty expect -")
```
Comment (SMillerDev) [inline Formula/jupp.rb:33]:
```suggestion
      pipe_output("env TERM=tty expect -", "spawn #{bin}/jupp;send \"q\";expect eof'")
```

Comment (johnsonjh) [reply Formula/jupp.rb]: Fixed in https://github.com/Homebrew/homebrew-core/pull/106097/commits/4c52a91b120a06e0982af237f7a649175fbc75a6.

File: Formula/jupp.rb (hunk 2 above)
Comment (johnsonjh) [reply Formula/jupp.rb]: Fixed in https://github.com/Homebrew/homebrew-core/pull/106097/commits/4981e23499c284236c96d7a8a66a0e36a2fcd888.

File: Formula/jupp.rb (hunk 1 above)
Comment (johnsonjh) [reply Formula/jupp.rb]: Fixed in https://github.com/Homebrew/homebrew-core/pull/106097/commits/2a9301c341462cd8063fe3ef2308d484566cca79.

File: Formula/jupp.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  depends_on "expect" => :test
+  uses_from_macos "ncurses"
+  on_macos do
+    depends_on "gnu-sed" => :build
+  end
+
+  def install
+    ENV.prepend_path "PATH", Formula["gnu-sed"].opt_libexec/"gnubin"
+    system "autoreconf", "-vfi"
+    system "sh", "./configure", *std_configure_args,
```
[the remark this answers is not in the document]
Comment (johnsonjh) [reply Formula/jupp.rb:22]: Yes, it does work.  I make that modification easily - I tested it both ways.

File: Formula/jupp.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  # Upstream HEAD in CVS: http://www.mirbsd.org/cvs.cgi/contrib/code/jupp/
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "expect" => :test
+  uses_from_macos "ncurses"
+  on_macos do
+    depends_on "gnu-sed" => :build
+  end
```
Comment (SMillerDev) [inline Formula/jupp.rb:17]:
```suggestion
  depends_on "expect" => :test

  uses_from_macos "ncurses"

  on_macos do
    depends_on "gnu-sed" => :build
  end
```

**Comment by SMillerDev** [comment]
Looks fine to me, but can you squash your work into 2 commits:
- `joe: conflict with jupp`
- `jupp 3.1jupp40 (new formula)`

**Comment by johnsonjh** [comment]
> Looks fine to me, but can you squash your work into 2 commits [...]

Done!

**Review by SMillerDev — APPROVED** [review]
Thanks @johnsonjh  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by johnsonjh** [comment]
@SMillerDev Do I need to squash this for you, or all set?

**Pull request closed without merging**